pyccel/codegen/printing/cucode.py

```diff
@@ -62,13 +61,46 @@ def _print_Module(self, expr):
                 local_imports += self._print(imp)
 
         imports = f'{c_headers_imports}\
-                    extern "C"{{\n\
+                   \n\
                     {local_imports}\
-                    }}'
+                    '
 
         code = f'{imports}\n\
                  {global_variables}\n\
                  {body}\n'
 
         self.exit_scope()
         return code
+
+    def _print_ModuleHeader(self, expr):
+        self.set_scope(expr.module.scope)
+        self._in_header = True
+        name = expr.module.name
+        if isinstance(name, AsName):
+            name = name.name
+
+        funcs = ""
+        cuda_headers = ''
+        for f in expr.module.funcs:
+            if not f.is_inline:
+                if 'kernel' in f.decorators:  # Checking for 'kernel' decorator
+                    cuda_headers += self.function_signature(f) + ';'
+                else:
+                    funcs += self.function_signature(f) + ';'
+        global_variables = ''.join(['extern '+self._print(d) for d in expr.module.declarations if not d.variable.is_private])
```

Comment:
Unnecessary memory allocation
```suggestion
        global_variables = ''.join('extern '+self._print(d) for d in expr.module.declarations if not d.variable.is_private)
```

Reply:
fixed 9eab7f3